[fix bug 1258303] Implement Yahoo! Search updates on /firstrun/learnmore

alexgibson opened the pull request:
## Description (needs to be live by 31 March)
- Adds 2 new `/firstrun/learnmore` templates that will be used via funnelcake builds of en-US browsers to test Yahoo! Search retention.
- Also moves the current `/firstrun/learnmore` to a more permanent folder location

## Bugzilla
https://bugzilla.mozilla.org/show_bug.cgi?id=1258303

## Testing
- Enable UITour for local development
- Template 1 URL: `/45.0/firstrun/learnmore/?f=64`
- Template 2 URL: `/45.0/firstrun/learnmore/?f=65`
- Clicking "Try it now" should open the search panel and populate the field with the text "Batman vs Superman"

## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing (none for this page).

alexgibson commented:
Still waiting on some GA requirements, but going to open this up as ready for general code review.

jpetto commented on bedrock/firefox/templates/firefox/firstrun/learnmore/yahoo/base.html:
```diff
@@ -0,0 +1,81 @@
+    <p class="pointer hidden">{{ _('Your search is ready!') }}</p>
+  </div>
+  <figure class="ravioli">
+    {{ high_res_img('firefox/firstrun/learnmore/yahoo/browser.png', {'alt': 'Yahoo! Search'}) }}
```
I see "Yahoo! Search" here, but "Yahoo Search" in the copy above. Is one preferred over the other? Can we use them interchangeably?

alexgibson replied:
Will change for consistency with the copy :+1:

jpetto commented on media/js/firefox/firstrun/learnmore/yahoo-search.js:
```diff
@@ -0,0 +1,93 @@
+            Promise.all([searchProviders, availableTargets]).then(function(results) {
+                var engines = results[0];
+                var targets = results[1];
+                resolve(targets && targets.indexOf('search') > -1 && engines && engines.indexOf('searchEngine-yahoo'));
```
Do we need a `> -1` check on `engines.indexOf('searchEngine-yahoo')`? If `searchEngine-yahoo` happened to be in position 0, could we get unexpected results here?

alexgibson replied:
Good catch, I totally omitted it by mistake

jpetto commented on bedrock/firefox/templates/firefox/firstrun/learnmore/yahoo/base.html:
```diff
@@ -0,0 +1,81 @@
+{% block content %}
+<main role="main" class="container">
+  {% block main_heading %}
+    <h1>{{ _('Search and see, with Yahoo Search.') }}</h1>
```
It's kind of a preference thing, but when very little is changing in the template (like just the heading), you could use a conditional and context data ([example](https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/views.py#L326-L349)) to create 1 HTML file instead of 3. Would also make implementing the winning headline a touch easier (fewer templates to shuffle around).

Just a suggestion for next time. :grin:

alexgibson replied:
Yeah, we originally had much more variation in the team played, but was cut down to just a single headline. You have convinced me to refactor nonetheless :smile:

jpetto commented:
A couple small questions, but looks great overall. :lollipop:  r+wc

alexgibson commented:
@jpetto - thanks for the review. Updated and added some tests for the view logic. Also switched to using `?f=64` and `?f=65` for the query param, as [requested in the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1258303#c16)

jpetto commented:
Looking good @alexgibson. I'd say a squarsh and merge is indeed in order.

alexgibson commented:
Thanks @jpetto squished!
